konduit-serving-core/src/main/java/ai/konduit/serving/launcher/KonduitServingLauncher.java

```diff
@@ -0,0 +1,153 @@
+            "inference", InferenceVerticle.class.getCanonicalName(),
+            "memmap", InferenceVerticle.class.getCanonicalName()
+    );
+
+    InferenceConfiguration inferenceConfiguration;
+
+    static {
+        LogUtils.setAppendersForCommandLine();
+
+        setProperty(LOGGER_DELEGATE_FACTORY_CLASS_NAME, SLF4JLogDelegateFactory.class.getName());
+        LoggerFactory.getLogger(LoggerFactory.class); // Required for Logback to work in Vertx
+
+        setProperty("vertx.cwd", new File(".").getAbsolutePath());
+        setProperty(CACHE_DIR_BASE_PROP_NAME, DEFAULT_FILE_CACHING_DIR);
+        setProperty(DISABLE_CP_RESOLVING_PROP_NAME, Boolean.TRUE.toString());
+    }
+
+    @Override
+    protected String getMainVerticle() {
+        return InferenceVerticle.class.getCanonicalName();
+    }
+
+    @Override
+    protected String getDefaultCommand() {
+        return ServeCommand.class.getAnnotation(Name.class).value();
+    }
+
+    /**
+     * Initializes the {@link VertxOptions} for deployment and use in a
+     * {@link Vertx} instance.
+     * The following other initialization also happens:
+     * {@code Vertx Working Directory} gets set (vertx.cwd) and {vertx.caseDirBase)
+     * (vertx.disableFileCPResolving) gets set to true
+     * (vertx.logger-delegate-factory-class-name) gets set to io.vertx.core.logging.SLF4JLogDelegateFactory
+     * The {@link MeterRegistry} field and associated prometheus configuration gets setup
+     * The {@link VertxOptions} event but options also get set
+     */
+    @Override
+    public void beforeStartingVertx(VertxOptions options) {
+        MicrometerMetricsOptions micrometerMetricsOptions = new MicrometerMetricsOptions()
+                .setMicrometerRegistry(new PrometheusMeterRegistry(PrometheusConfig.DEFAULT))
+                .setPrometheusOptions(new VertxPrometheusOptions()
+                        .setEnabled(true));
+
+        log.info("Setup micro meter options.");
+        BackendRegistries.setupBackend(micrometerMetricsOptions);
+
+        options.setMetricsOptions(micrometerMetricsOptions);
+    }
+
+    @Override
+    public void afterStartingVertx(Vertx vertx) {
+        vertx.registerVerticleFactory(new VerticleFactory() {
+
+            @Override
+            public String prefix() {
+                return KONDUIT_PREFIX;
+            }
+
+            @Override
+            public Verticle createVerticle(String verticleName, ClassLoader classLoader) throws ClassNotFoundException, IllegalAccessException, InstantiationException {
+                String serviceType = verticleName.replace(KONDUIT_PREFIX + ":", "");
+
+                if(services.containsKey(serviceType)) {
+                    return (Verticle) ClassLoader.getSystemClassLoader().loadClass(services.get(serviceType)).newInstance();
+                } else {
+                    log.error(String.format("Invalid service type %s. Possible values are: ", Arrays.asList(services.keySet())));
```

Comment:
Missing service type, don't need to create list, don't use String.format
```
log.error("Invalid service type \"{}\". Possible values are: {}", serviceType, services.keySet());
```

Reply:
Oh my bad...